dbms/src/Common/NamePrompter.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <Core/Types.h>
+
+#include <cctype>
+#include <algorithm>
+#include <queue>
+#include <utility>
+
+#include <iostream>
+
+namespace DB
+{
+
+template <size_t MistakeFactor, size_t MaxNumHints>
+class NamePrompter
+{
+public:
+    using DistanceIndex = std::pair<size_t, size_t>;
+    using DistanceIndexQueue = std::priority_queue<DistanceIndex>;
+
+    static std::vector<String> getHints(const String & name, const std::vector<String> & prompting_strings)
+    {
+        DistanceIndexQueue queue;
+        for (size_t i = 0; i < prompting_strings.size(); ++i)
+            appendToQueue(i, name, queue, prompting_strings);
+        return release(queue, prompting_strings);
+    }
+
+private:
+
+    static size_t LevenshteinDistance(const String & lhs, const String & rhs)
```

Comment:
and functions are in camelCase with lowercase initial letter.

Reply:
Levenshtein is a name :) I was doubting